### ISSUE TITLE
Unable connect to server

### ISSUE THREAD

**Issue by odin-linux** [magefree/mage#9137]
I am able to host a local server and connect to it.

1. Open launcher: `java -jar XMageLauncher-0.3.8.jar`.   
```
➜  ~ java -version
openjdk version "1.8.0_332"
OpenJDK Runtime Environment (build 1.8.0_332-b09)
OpenJDK 64-Bit Server VM (build 25.332-b09, mixed mode)

Reading config from http://xmage.today//config.json
XMage folder:  <path>
XMage Launcher version installed:  0.3.8
XMage Launcher version available:  0.3.8
Java version installed:  1.8.0_201
Java version available:  1.8.0_201
XMage version installed:  1.4.51-dev (2022-06-17 05-05)
XMage version available:  1.4.51-dev (2022-06-17 05-05)

```
2. Launch Client
3. Select us.xmage.today:17171
- Username populated
- Automatically connect... off
- Force update... off
- Server is online
4. Press Connect: Unable connect to server
```
INFO  2022-06-22 00:25:31,047 Starting MAGE client version 1.4.51-V4-beta5 (build: 2022-06-17 04:58)                     =>[main] MageFrame.main 
INFO  2022-06-22 00:25:31,048 Logging level: INFO                                                                        =>[main] MageFrame.main 
INFO  2022-06-22 00:25:31,048 Default charset: UTF-8                                                                     =>[main] MageFrame.main 
INFO  2022-06-22 00:25:31,941 Using GUI theme: Default                                                                   =>[AWT-EventQueue-0] PreferencesDialog.getCurrentTheme 
INFO  2022-06-22 00:25:31,949 Loading database...                                                                        =>[AWT-EventQueue-0] RepositoryUtil.bootstrapLocalDb 
INFO  2022-06-22 00:25:32,266 Images: search broken files...                                                             =>[AWT-EventQueue-0] MageFrame.<init> 
INFO  2022-06-22 00:25:32,478 Loading symbols...                                                                         =>[AWT-EventQueue-0] ManaSymbols.loadImages 
INFO  2022-06-22 00:25:32,881 Loading plugins...                                                                         =>[AWT-EventQueue-0] Plugins.loadPlugins 
INFO  2022-06-22 00:25:32,909 [Game counter plugin, version 0.1] has been loaded.                                        =>[AWT-EventQueue-0] CounterPluginImpl.newPlugin 
INFO  2022-06-22 00:25:32,916 [Card plugin, version 0.7] has been loaded.                                                =>[AWT-EventQueue-0] CardPluginImpl.newPlugin 
INFO  2022-06-22 00:25:32,917 [Theme plugin, version 0.5] has been loaded.                                               =>[AWT-EventQueue-0] ThemePluginImpl.newPlugin 
INFO  2022-06-22 00:25:32,918 Done.                                                                                      =>[AWT-EventQueue-0] Plugins.loadPlugins 
INFO  2022-06-22 00:25:33,578 Client start up time: 2 seconds                                                            =>[AWT-EventQueue-0] MageFrame.lambda$new$2 
INFO  2022-06-22 00:26:48,169 Connect: connecting to server us.xmage.today:17171                                         =>[SwingWorker-pool-5-thread-4] SessionImpl$5.work 
WARN  2022-06-22 00:26:48,534 Cannot connect                                                                             =>[SwingWorker-pool-5-thread-3] SessionImpl.handleCannotConnectException 
org.jboss.remoting.CannotConnectException: Error setting up client lease upon performing connect.
	at org.jboss.remoting.Client.connect(Client.java:1928)
	at org.jboss.remoting.Client.connect(Client.java:737)
	at mage.remote.SessionImpl$5.work(SessionImpl.java:429)
	at mage.remote.SessionImpl$RemotingTask$1.doInBackground(SessionImpl.java:95)
	at mage.remote.SessionImpl$RemotingTask$1.doInBackground(SessionImpl.java:91)
	at javax.swing.SwingWorker$1.call(SwingWorker.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at javax.swing.SwingWorker.run(SwingWorker.java:334)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.Exception: Error setting up client lease
	at org.jboss.remoting.MicroRemoteClientInvoker.establishLease(MicroRemoteClientInvoker.java:563)
	at org.jboss.remoting.Client.setupClientLease(Client.java:2070)
	at org.jboss.remoting.Client.connect(Client.java:1924)
	... 10 more
Caused by: org.jboss.remoting.CannotConnectException: Can not get connection to server. Problem establishing socket connection for InvokerLocator [bisocket://us.xmage.today:17171/?onewayThreadPool=mage.remote.CustomThreadPool&serializationtype=java]
	at org.jboss.remoting.transport.socket.MicroSocketClientInvoker.transport(MicroSocketClientInvoker.java:855)
	at org.jboss.remoting.transport.bisocket.BisocketClientInvoker.transport(BisocketClientInvoker.java:470)
	at org.jboss.remoting.MicroRemoteClientInvoker.invoke(MicroRemoteClientInvoker.java:169)
	at org.jboss.remoting.MicroRemoteClientInvoker.establishLease(MicroRemoteClientInvoker.java:529)
	... 12 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.jboss.remoting.transport.socket.SocketClientInvoker.createClientSocket(SocketClientInvoker.java:175)
	at org.jboss.remoting.transport.socket.MicroSocketClientInvoker.getConnection(MicroSocketClientInvoker.java:1220)
	at org.jboss.remoting.transport.socket.MicroSocketClientInvoker.transport(MicroSocketClientInvoker.java:833)
	... 15 more
Caused by: java.io.StreamCorruptedException: invalid stream header: 6A627331
	at java.io.ObjectInputStream.readStreamHeader(ObjectInputStream.java:866)
	at java.io.ObjectInputStream.<init>(ObjectInputStream.java:358)
	at org.jboss.remoting.loading.ObjectInputStreamWithClassLoader.<init>(ObjectInputStreamWithClassLoader.java:100)
	at org.jboss.remoting.serialization.impl.java.JavaSerializationManager.createInput(JavaSerializationManager.java:54)
	at org.jboss.remoting.marshal.serializable.SerializableUnMarshaller.getMarshallingStream(SerializableUnMarshaller.java:75)
	at org.jboss.remoting.marshal.serializable.SerializableUnMarshaller.getMarshallingStream(SerializableUnMarshaller.java:58)
	at org.jboss.remoting.transport.socket.ClientSocketWrapper.createInputStream(ClientSocketWrapper.java:206)
	at org.jboss.remoting.transport.socket.ClientSocketWrapper.createStreams(ClientSocketWrapper.java:189)
	at org.jboss.remoting.transport.socket.ClientSocketWrapper.<init>(ClientSocketWrapper.java:67)
	... 22 more
INFO  2022-06-22 00:26:48,549 Disconnecting DONE                                                                         =>[SwingWorker-pool-5-thread-3] SessionImpl.disconnect 
INFO  2022-06-22 00:26:48,550 Disconnected from user mode                                                                =>[SwingWorker-pool-5-thread-3] MageFrame.disconnected 
```

**Comment by Alex-Vasile** [comment]
Are you able to connect to a local server?

**Comment by Alex-Vasile** [comment]
Does disabling the firewall on your computer solve the issue?

**Comment by odin-linux** [comment]
I've attempted to connect without my firewall running and have not been successful at connecting.

**Comment by Alex-Vasile** [comment]
> I've attempted to connect without my firewall running and have not been successful at connecting.

I don't know what to say :/

Does the stable (non-beta) version connect?

**Comment by the-red-lily** [comment]
I had similar issue when I downloaded the jars individually, and the problem was solved when I downloaded the launcher from the website. But if that's what you did it isn't any help :/

**Comment by JayDi85** [comment]
> invalid stream header: 6A627331

It's about incompatible versions (wrong data in the serialization). I think it's a java serialization problem (xmage's engine has not changed recently). Can be related to bug/changes in the #8627

1. Try normal release from the xmage.de -- is it working (can you connect to de-server)?
2. Try to downgrade your OpenJDK version (see [here for details](https://github.com/magefree/mage/issues/8470#issuecomment-980830855)).

**Comment by Alex-Vasile** [comment]
Rereading your initial post, it looks like you're trying to connect to the non-beta server with a beta client:
> XMage version installed:  1.4.51-dev (2022-06-17 05-05)
> XMage version available:  1.4.51-dev (2022-06-17 05-05)

> Select us.xmage.today:17171

The beta client is for the beta server only. Try connectiong to beta.xmage.today with the beta client
